### ISSUE TITLE
Can't reach my APIs after update to 2.1.0-beta3

### ISSUE THREAD

**Issue by mizrachiran** [Microsoft/aspnet-api-versioning#115]
Hi,

I updated my nugets to use the latest pre-release versions today.
i'm using the following packages:

  <package id="Microsoft.AspNet.OData" version="6.0.0" targetFramework="net451" />
  <package id="Microsoft.AspNet.OData.Versioning" version="2.1.0-beta3" targetFramework="net451" />
  <package id="Microsoft.Data.Edm" version="5.6.4" targetFramework="net451" />
  <package id="Microsoft.Data.OData" version="5.6.4" targetFramework="net451" />
  <package id="Microsoft.OData.Core" version="7.1.1" targetFramework="net451" />
  <package id="Microsoft.OData.Edm" version="7.1.1" targetFramework="net451" />

I used to have the url convention set to ODataSimplified as follow:
            **config.SetUrlConventions(ODataUrlConventions.ODataSimplified);**
Since this extension method is no longer available on the new versions, i can't reach my APIs anymore in the way i used to, not sure if related to the API versioning package.

I configure my route as follow:

            configuration.MapVersionedODataRoutes(
                "odata",
                "api/v{apiVersion}",
                models,
                builder =>
                    builder
                        .AddService(ServiceLifetime.Singleton, s => new CaseInsensitiveODataUriResolver()),
                new DefaultODataBatchHandler(httpServer));

and the URLs i'm trying:
/api/v1.0/files/123
/api/v1.0/files(123)
/api/v1.0/Files(123)
/api/v1.0/Files/123

non of the above is working (used to work before the update).

Any ideas?

Thanks!!!

more settings:

config.AddApiVersioning(o =>
            {
                o.ReportApiVersions = true;
                o.AssumeDefaultVersionWhenUnspecified = true;
                o.DefaultApiVersion = new ApiVersion(1, 0);
                o.ApiVersionReader = ApiVersionReader.Combine(
                    new QueryStringApiVersionReader(),
                    new HeaderApiVersionReader("api-version", "x-ms-version"));
                o.Conventions.Controller<FilesController>()
                    .HasApiVersion(new ApiVersion(1, 0))
                    .HasApiVersion(new ApiVersion(2, 0));
            });

       var modelBuilder = new VersionedODataModelBuilder(config)
            {
                ModelBuilderFactory = () => new ODataConventionModelBuilder().EnableLowerCamelCase()
            };

**Comment by commonsensesoftware** [comment]
Several things changed in the OData 6.0.0 release. Many of them caused internal, but breaking changes for API versioning (which is why it's taken so long to update). The simplified URL conventions can be still be set, but it has to be done with dependency injection AFAIK. Try looking at the documentation [here](http://odata.github.io/WebApi/#13-04-DependencyInjection) and [here](http://odata.github.io/odata.net/v7/#01-05-di-support). I believe you want to replace either **UriPathParser** and/or **IODataPathHandler**.

I would expect that 2 and 3 of your listed URLs to still work. Are you using a custom **ITraceWriter** implementation by chance? An issue recently came up that I'm investigating related to decorated types generated by Web API when you register tracing.

I don't any of that will solve your issue, but it might give you some clues. If you're able to strip things down to full configuration setup and an empty controller with your conventions, go ahead and attach it. A repro goes a long way in troubleshooting an issue.

Thanks

**Comment by mizrachiran** [comment]
I still can't reach my service APIs using the key-as-segment convention.
I would appreciate if you can point me to a working sample.


Thanks!!!

**Comment by commonsensesoftware** [comment]
Hmmm... I think you've provided enough for me to create a simple repro. I will see if I can't put together the world's simplest, working example and then you can compare it to whatever you have.

**Comment by commonsensesoftware** [comment]
It seems your question is more about how to configure the OData library to use simple URL conventions in 6.0+. There doesn't appear to be anything specifically related to API versioning. In order to enable the equivalent of the old simple URL convention, you need to add the following to the service configuration:

```c#
builder.AddService(
    ServiceLifetime.Singleton,
    typeof( IODataPathHandler ),
    sp => new DefaultODataPathHandler() { UrlKeyDelimiter = ODataUrlKeyDelimiter.Slash } );
```

I tested it and things worked exactly as expected. If you want to reproduce an end-to-end scenario for yourself, you can take the [basic OData example](../tree/master/samples/webapi/BasicODataWebApiSample) and add this one line to the **ConfigureODataServices** method in the **Startup.cs** file (line 45). Everything else just works after that. Interestingly, the OData route attributes continue to work for simple key segments, even when the template is defined as `({key})`.

If there was an issue somehow caused by API versioning, it has seemingly been fixed in the 1.1 RC. I hope that helps. Let me know if you have any other questions.

**Comment by mizrachiran** [comment]
Thanks you so much!!! you made my day!!!
